### ISSUE TITLE
How can we implement a image-in-and-image-out CNN?

### ISSUE THREAD

**Issue by alzhusan** [fchollet/keras#359]
In visual object detection task, we should label each pixel with some labeles such as 'road', 'car', 'tree' and so on, as in the figure below.

![segmentation](https://cloud.githubusercontent.com/assets/3699422/8560760/fb516648-254d-11e5-8b40-62b3fabffcec.png)
 left: the input image; right: the target.

I think this task can be solved with convolutional neural network (CNN), which shows its power in many image-related tasks.

In the examples folder, the CNN expamples are all about image labelling task which input is a image and output is a single label. How can we employ CNN to the visual object detection task, which requires the output is also a image?

**Comment by pranv** [comment]
The output of such networks are not images. It's obtained by Image Segmentation, or in other words each pixel is labelled. There are many ways to do it, all of which could be built on top of Keras.

**Comment by dongb5** [comment]
I think you do not need an image-output. The output should be a matrix with the same size of the input image and each element in the matrix is a label.

**Comment by Ori226** [comment]
There is an interesting article from Yann LeCun about employing CNN for scene labeling:
http://yann.lecun.com/exdb/publis/pdf/farabet-pami-13.pdf

**Comment by dk683** [comment]
Hello! Would there be a reference that I could look for the image segmentation by Keras?
I am trying to achieve a similar task, but I am having some difficulty.
Thank you!
